lib/getBinaryGltf.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var sizeOf = require('image-size');
+var mime = require('mime');
+var mergeBuffers = require('./mergeBuffers');
+var removePipelineExtras = require('./removePipelineExtras');
+
+module.exports = getBinaryGltf;
+
+function getBinaryGltf(gltf, callback) {
+    //Create the special binary buffer from the existing buffers
+    gltf.bufferViews = defaultValue(gltf.bufferViews, {});
+    gltf.buffers = defaultValue(gltf.buffers, {});
+    mergeBuffers(gltf, 'binary_glTF');
+
+    var bufferViews = gltf.bufferViews;
+    var buffers = gltf.buffers;
+    var currentOffset = buffers.binary_glTF.byteLength;
+    var body = buffers.binary_glTF.extras._pipeline.source;
+    var currentBinaryView = 0;
+    //Update object with KHR_binary_glTF properties and add to body and bufferViews
+    function updateBinaryObject(name) {
```

Comment:
I think it is better to put this function outside of the outer `getBinaryGltf` function and just pass in any data needs.  I expect it will get optimized better.